patches/server/0724-Use-getChunkIfLoadedImmediately-in-places.patch

```diff
@@ -0,0 +1,62 @@
+-        return this.chunkSource.getChunk(x, z, false);
++        return this.chunkSource.getChunkAtIfLoadedImmediately(x, z); // Paper
+     }
+ 
+     // Paper start - Asynchronous IO
+diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+index 8708eb173fbb17cf84a0c6d9a42623750472f650..35fa416a8ce332e823ed5077a8fd3492683d7ad0 100644
+--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -1318,7 +1318,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
+                                 speed = this.player.getAbilities().walkingSpeed * 10f;
+                             }
+                             // Paper start - Prevent moving into unloaded chunks
+-                            if (player.level.paperConfig.preventMovingIntoUnloadedChunks && (this.player.getX() != toX || this.player.getZ() != toZ) && !worldserver.hasChunk((int) Math.floor(toX) >> 4, (int) Math.floor(toZ) >> 4)) {
++                            if (player.level.paperConfig.preventMovingIntoUnloadedChunks && (this.player.getX() != toX || this.player.getZ() != toZ) && worldserver.getChunkIfLoadedImmediately((int) Math.floor(toX) >> 4, (int) Math.floor(toZ) >> 4) == null) { // Paper - use getIfLoadedImmediately
+                                 this.internalTeleport(this.player.getX(), this.player.getY(), this.player.getZ(), this.player.getYRot(), this.player.getXRot(), Collections.emptySet(), true);
+                                 return;
+                             }
+diff --git a/src/main/java/net/minecraft/world/level/Level.java b/src/main/java/net/minecraft/world/level/Level.java
+index 1c171599dbde44389013dc08ca8263742d58d721..4eb2994d616511e197c1fd0273d9379500070bd0 100644
+--- a/src/main/java/net/minecraft/world/level/Level.java
++++ b/src/main/java/net/minecraft/world/level/Level.java
+@@ -191,6 +191,13 @@ public abstract class Level implements LevelAccessor, AutoCloseable {
+         return (CraftServer) Bukkit.getServer();
+     }
+ 
++    // Paper start
++    @Override
++    public boolean hasChunk(int chunkX, int chunkZ) {
++        return ((ServerLevel)this).getChunkIfLoaded(chunkX, chunkZ) != null;
++    }
++    // Paper end
++
+     public ResourceKey<DimensionType> getTypeKey() {
+         return this.typeKey;
+     }
+@@ -1366,7 +1373,7 @@ public abstract class Level implements LevelAccessor, AutoCloseable {
+ 
+         for (int l1 = j; l1 <= l; ++l1) {
+             for (int i2 = k; i2 <= i1; ++i2) {
+-                LevelChunk chunk = this.getChunkSource().getChunkNow(l1, i2);
++                LevelChunk chunk = (LevelChunk)this.getChunkIfLoadedImmediately(i1, j1); // Paper
```

Comment:
```suggestion
+                LevelChunk chunk = (LevelChunk) this.getChunkIfLoadedImmediately(l1, i2); // Paper
```

Reply:
smh leaf

Reply:
not leaf, you just didn't port it to the new the mappings correctly :p

Reply:
It was already ported, I'll get it fixed

Reply:
it was auto-remapped, which doesn't always get variable names correct, but I digress.

Reply:
I missed the var names myself, I only saw the missing formatting whoops.